src/app/test/db/mongo/util_test.py

```diff
@@ -331,3 +299,48 @@ def test_ensure_local_garden_updates_garden_from_config(self, monkeypatch):
         garden = Garden.objects.get(connection_type="LOCAL")
 
         assert garden.name == config.get("garden.name")
+
+
+class TestEnsureRoles:
+    @pytest.fixture
+    def role_definition_yaml(self):
+        role_list = [{"name": "testrole1", "permissions": ["garden:read"]}]
+        yield yaml.dump(role_list)
+
+    @classmethod
+    def setup_class(cls):
+        connect("beer_garden", host="mongomock://localhost")
+
+    def teardown_method(self):
+        beer_garden.db.mongo.models.Role.drop_collection()
+
+    def config_get_value(self, config_name):
+        return "/some/file/path"
+
+    def config_get_none(self, config_name):
+        return None
+
+    def test_ensure_roles_creates_roles_defined_in_file(
+        self, monkeypatch, role_definition_yaml
+    ):
+        """ensure_roles should create the roles defined in the auth.role_definition_file
+        if specified"""
+        monkeypatch.setattr(config, "get", self.config_get_value)
+
+        role_definition_file = config.get("auth.role_definition_file")
+
+        with patch(
+            "builtins.open", mock_open(read_data=role_definition_yaml)
+        ) as mock_file_read:
+            ensure_roles()
+            mock_file_read.assert_called_with(role_definition_file, "r")
+
+        assert len(Role.objects.filter(name="testrole1")) == 1
```

Comment:
Ideally the `sync_roles` call would be mocked here and I would just check that it got called (and not called in the next test), but I couldn't for the life of me get that mock to work.  I'm guessing there's something about the interplay between `pytest` and `patch` I'm not understanding. Either way I gave up and just resorted to letting it actually execute `sync_roles`, even though it kind of amounts to a duplicate of tests dedicated to that function.